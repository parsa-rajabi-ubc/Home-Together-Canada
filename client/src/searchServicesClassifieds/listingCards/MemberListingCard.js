/**
 *
 * @Author:     Jeff Hatton
 * @Created:    2021.2.18
 *
 * @Description: Member Listing Card
 *
 */

import React from 'react';
import PropTypes from "prop-types";
import Moment from 'react-moment';
import {FaDog as Dog} from "react-icons/fa";
import {GiCigarette as Cigarette} from "react-icons/gi";
import Tooltip from "../../common/forms/Tooltip";

function MemberListingCard(props) {
    const {title, monthlyCost, petFriendly, smokeFriendly, shortDescription, datePosted} = props;

    return (
        <section className={"card-container mt-0"}>

            <div className={"inline ml-6 align-middle "}>
                <label className={"font-semibold justify-between"}> {title}</label>,
                <label className={"px-1"}>${monthlyCost} </label>

                <section className={"inline float-right pr-6 justify-end"}>
                    <Moment format="MMM D, YYYY">{datePosted}</Moment>
                </section>
            </div>

            <div className={"ml-6 mt-4 flex"}>
                <label className={"w-full font-light"}> {shortDescription} </label>

                <section className={"flex pr-6 "}>
                    {petFriendly && <Tooltip text={"Pet Friendly"} toolTipID={"PetFriendly"}
                                             customIcon={<Dog className={"text-blue-400 ml-1"}/>}/>}

                    {smokeFriendly && <Tooltip text={"Smoke Friendly"} toolTipID={"SmokeFriendly"}
                                               customIcon={<Cigarette className={"text-red-500 ml-1"}/>}/>}
                </section>
            </div>

        </section>
    );
}

MemberListingCard.propTypes = {
    title: PropTypes.string.isRequired,
    monthlyCost: PropTypes.number.isRequired,
    petFriendly: PropTypes.bool.isRequired,
    smokeFriendly: PropTypes.bool.isRequired,
    shortDescription: PropTypes.string.isRequired,
    datePosted: PropTypes.string.isRequired,
}


export default MemberListingCard;