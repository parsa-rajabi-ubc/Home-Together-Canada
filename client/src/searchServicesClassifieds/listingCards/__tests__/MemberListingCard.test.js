/**
 * @Author:     Jeff Hatton
 * @Created:    2021.2.19
 *
 * @Description: Member Listing Card Test
 *
 */

import React from 'react';
import ShallowRenderer from "react-test-renderer/shallow";
import MemberListingCard from "../MemberListingCard";

describe('MemberListingCard', () => {
    describe('Container test', () => {
        it('should match snapshot test', () => {
            //given
            const props = {
                    title: "Rooms for Rent",
                    shortDescription: "3 Furnished Rooms, for students preferably",
                    monthlyCost: 650,
                    petFriendly: false,
                    smokeFriendly: true,
                    datePosted: "2021-02-14T07:42:39.000Z"
                }

            //when
            const renderer = new ShallowRenderer();
            renderer.render(<MemberListingCard {...props}/>);
            const result = renderer.getRenderOutput();
            //then
            expect(result).toMatchSnapshot();
        });
    });
});