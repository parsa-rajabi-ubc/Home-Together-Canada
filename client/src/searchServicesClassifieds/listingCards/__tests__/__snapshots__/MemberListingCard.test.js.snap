// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberListingCard Container test should match snapshot test 1`] = `
<section
  className="card-container mt-0"
>
  <div
    className="inline ml-6 align-middle "
  >
    <label
      className="font-semibold justify-between"
    >
       
      Rooms for Rent
    </label>
    ,
    <label
      className="px-1"
    >
      $
      650
       
    </label>
    <section
      className="inline float-right pr-6 justify-end"
    >
      <n
        ago={false}
        calendar={false}
        decimal={false}
        element={null}
        filter={[Function]}
        format="MMM D, YYYY"
        fromNow={false}
        interval={60000}
        local={false}
        onChange={[Function]}
        titleFormat=""
        toNow={false}
        trim={false}
        unit={null}
        unix={false}
        utc={false}
        withTitle={false}
      >
        2021-02-14T07:42:39.000Z
      </n>
    </section>
  </div>
  <div
    className="ml-6 mt-4 flex"
  >
    <label
      className="w-full font-light"
    >
       
      3 Furnished Rooms, for students preferably
       
    </label>
    <section
      className="flex pr-6 "
    >
      <Tooltip
        customIcon={
          <GiCigarette
            className="text-red-500 ml-1"
          />
        }
        text="Smoke Friendly"
        toolTipID="SmokeFriendly"
      />
    </section>
  </div>
</section>
`;
